roadmap/implementers-guide/src/node/disputes/participation.md

```diff
@@ -0,0 +1,106 @@
+# Dispute Participation
+
+Tracks all open disputes in the active leaves set.
+
+## Remarks
+
+`UnsignedTransaction`s are OK to be used, since the inner
+element `CommittedCandidateReceipt` is verifiable.
+
+## Assumptions
+
+* If each validator gossips its own vote regarding the disputed block, upon receiving the first vote (the initiating dispute vote) there is only
+a need for a very small gossip only containing the `Vote` itself.
+
+## IO
+
+Inputs:
+
+* `DisputeGossip::Vote` (votes of other validators)
+* `DisputeParticipationMessage::Detection`
+* `DisputeParticipationMessage::Resolution`
+
+Outputs:
+
+* `AvailabilityRecoveryMessage::RecoverAvailableData`
+* `CandidateValidationMessage::ValidateFromChainState`
+* `DisputeGossip::Vote` (own vote)
+
+## Messages
+
+```rust
+enum DisputeParticipationMessage {
+    Detection {
+        candidate: CandidateHash,
+        votes: HashMap<ValidatorId, Vote>,
+    },
+
+    Resolution{
+        resolution: Resolution,
+        votes: HashMap<ValidatorId, Vote>,
+    },
+}
+```
+
+Distribution of our own vote via gossip, but also
+receive them in order to store it to `VotesDB`:
+
+```rust
+enum DisputeGossip {
+    /// A vote by a validator, referenced by session
+    /// index and validator index
+    Vote(SessionIndex, ValidatorIndex, Vote),
+}
+```
+
+## Helper Structs
+
+```rust
+/// Resolution of a vote:
+enum Resolution {
+    /// The block is valid
+    Valid,
+    /// The block is valid
+    Invalid,
+}
+```
+
+## Session Change
+
+> TODO
+
+## Storage
+
+Nothing to store, everything lives in `VotesDB`.
+
+## Sequence
+
+### Detection
+
+1. Receive `DisputeParticipationMessage::Detection`
+1. Request `AvailableData` via `RecoverAvailableData` to obtain the `PoV`.
+1. Call `ValidateFromChainState` to validate the `PoV` with the `CandidateDescriptor`.
+    1. Cast vote according to the `ValidationResult`.
+    1. Store our vote to the `VotesDB` via `VotesDBMessage::StoreVote`
+    1. Gossip our vote on the network.
+
+### Resolution
+
+1. Receive `DisputeParticipationMessage::Resolution`
+1. Craft an unsigned transaction with all votes cast
+  1. includes `CandidateReceipts`
+  1. includes `Vote`s
+1. Delete all relevant data associated with this dispute
```

Comment:
We do not care overly if `CandidateReceipts` lives on the current chain, although if it does then we could impact the fork choice rule later.

Reply:
Right, we always craft the unsigned transaction to all active heads.